code/modules/maps/template_types/random_exoplanet/random_planet_level_data.dm

```diff
@@ -82,11 +82,12 @@
 	//Rename the surface area if we have one yet
 	adapt_location_name(parent_planetoid.name)
 
-///If we're getting atmos from our parent planet, decide if we're going to apply it, or ignore it
+///If we're getting atmos from our parent planet, apply it.
 /datum/level_data/planetoid/proc/apply_planet_atmosphere(var/datum/planetoid_data/P)
-	if(istype(exterior_atmosphere))
-		return //level atmos takes priority over planet atmos
-	exterior_atmosphere = P.atmosphere.Clone() //Make sure we get one instance per level
+	if(istype(P))
+		exterior_atmosphere = P.atmosphere.Clone()
```

Comment:
For randomized exoplanets it seems to always create one with `/datum/planetoid_data/random/proc/generate_atmosphere()`. It's possible for non-random exoplanets to have it be null if they didn't supply a preset composition, but a quick examination at the code suggests that there's some other planet stuff like `setup_flora()` expects a gas mixture datum and has no null checks, so a null atmosphere would probably break stuff like that as well.
I'll still put a check here however since the rest of the exoplanet code seems to have null checks for atmosphere.